Created on Wed 24th Feb 2021 16:26

@author: Drishika Nadella


In [25]:
import numpy as np

In [26]:
def orbit(T):
    """
    This function calculates the orbit of a satellite (r - R) given its time period T.
    """
    G = 6.67*10**(-11)
    M = 5.972*10**24
    R = 6371*10**3
    
    r = (G*M*T**2/(4*np.pi**2))**(1/3)
    return r-R

In [27]:
period = input("Enter the time period of the satellite in minutes: ")
h = orbit(int(period)*60)*10**(-3)
print("The height of the satellite is %7.3f km"% (h))

Enter the time period of the satellite in minutes: 45
The height of the satellite is -2181.092 km


### For a geosynchronous satellite:

Enter the time period of the satellite in minutes: 1440

The height of the satellite is 35860.625 km

### For a satellite with time period 90 minutes:

Enter the time period of the satellite in minutes: 90

The height of the satellite is 280.064 km

### For a satellite with time period 45 minutes: 

Enter the time period of the satellite in minutes: 45

The height of the satellite is -2181.092 km

The answer is in negatives because 45 minutes is too small a period for a satellite to orbit. The minimum height for a satellite is 100 km, called the Karman line.

### Question 2c

A sidereal day is about 4 minutes lesser than a normal solar day because a sidereal day is the time required for the Earth to complete one rotation with respect to fixed stars.

In [29]:
h_geo = orbit(24*60)*10**(-3)
h_sid = orbit(23.93*60)*10**(-3)
print("The height difference between a satellite placed in the geostationary orbit to the sidereal orbit is: %3.2f"% (h_geo - h_sid))

The height difference between a satellite placed in the geostationary orbit to the sidereal orbit is: 5.36
